sdk/servicebus/service-bus/src/core/messageSender.ts

```diff
@@ -92,119 +91,77 @@ export class MessageSender extends LinkEntity {
     this._retryOptions = normalizeRetryOptions(retryOptions);
     this._onAmqpError = (context: EventContext) => {
       const senderError = context.sender && context.sender.error;
-      if (senderError) {
-        const err = translate(senderError);
-        log.error(
-          "[%s] An error occurred for sender '%s': %O.",
-          this._context.namespace.connectionId,
-          this.name,
-          err
-        );
-      }
+      log.error(
+        "[%s] 'sender_error' event occurred on the sender '%s' with address '%s'. " +
+          "The associated error is: %O",
+        this._context.namespace.connectionId,
+        this.name,
+        this.address,
+        senderError
+      );
+      // TODO: Consider rejecting promise in trySendBatch() or createBatch()
```

Comment:
Is there an issue for this, or are you doing this as part of this PR?

Reply:
**Thanks, I wanted to ask about this, which is why I left it in the first place.**

This TODO is straight from the event-hubs, I'm not exactly sure what was meant or what was concluded over there.
https://github.com/Azure/azure-sdk-for-js/blob/0f2bf8e3e8d0b9d0d5a33857b5f1ee2a99928bf4/sdk/eventhub/event-hubs/src/eventHubSender.ts#L101
Reference from the PR - https://github.com/Azure/azure-sdk-for-js/pull/6737#discussion_r362901116
Whatever was decided for event-hubs would be applicable here too.

At this point, I guess I need more inputs on this.

Reply:
Gotcha. My vague recollection is that it seemed like a good idea, but if we haven't done it in event hubs yet either, I wouldn't focus on it for this PR then.

Reply:
Sounds good, Ramya suggested the same, logged an issue https://github.com/Azure/azure-sdk-for-js/issues/8656